Ghidra/Processors/RISCV/data/languages/riscv.zfh.sinc

```diff
@@ -0,0 +1,249 @@
+{
+	#TODO  redo this
+	local tmpfrs1 = frs1S;
+	local tmpfrs2 = frs2S;
+	fassignS(frd, tmpfrs1);
+	if (nan(tmpfrs1) && nan(tmpfrs2)) goto inst_next;
+	if (nan(tmpfrs2)) goto inst_next;
+	fassignS(frd, tmpfrs2);
+	if (nan(tmpfrs1)) goto inst_next;
+	if (tmpfrs2 f<= tmpfrs1) goto inst_next;
+	fassignS(frd, tmpfrs1);
+}
+
+:fmax.h frd,frs1S,frs2S is frs1S & frd & frs2S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x1 & funct7=0x16
+{
+	#TODO  redo this
+	local tmpfrs1 = frs1S;
+	local tmpfrs2 = frs2S;
+	fassignS(frd, tmpfrs1);
+	if (nan(tmpfrs1) && nan(tmpfrs2)) goto inst_next;
+	if (nan(tmpfrs2)) goto inst_next;
+	fassignS(frd, tmpfrs2);
+	if (nan(tmpfrs1)) goto inst_next;
+	if (tmpfrs2 f>= tmpfrs1) goto inst_next;
+	fassignS(frd, tmpfrs1);
+}
+
+:fsqrt.h frd,frs1S,FRM is frs1S & frd & FRM & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct7=0x2e & op2024=0x0
+{
+	local tmp:4 = sqrt(frs1S);
+	fassignS(frd, tmp);
+}
+
+:fsgnj.h frd,frs1S,frs2S is frs1S & frd & frs2S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x0 & funct7=0x12
+{
+	local tmp:$(SFLEN) = frs1S;
+	tmp[31,1] = frs2S[31,1];
+	fassignS(frd, tmp);
+}
+
+:fsgnjn.h frd,frs1S,frs2S is frs1S & frd & frs2S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x1 & funct7=0x12
+{
+	local tmp:$(SFLEN) = frs1S;
+	tmp[31,1] = !frs2S[31,1];
+	fassignS(frd, tmp);
+}
+
+:fsgnjx.h frd,frs1S,frs2S is frs1S & frd & frs2S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x2 & funct7=0x12
+{
+	local tmp:$(SFLEN) = frs1S;
+	tmp[31,1] = tmp[31,1] ^ frs2S[31,1];
+	fassignS(frd, tmp);
+}
+
+:fmv.h frd,frs1S is frd & frs1S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x0 & funct7=0x12 & fr1519=fr2024
+{
+	fassignS(frd, frs1S);
+}
+
+:fneg.h frd,frs1S is frs1S & frd & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x1 & funct7=0x12 & op1519=op2024
+{
+	local tmp:4 = f- frs1S;
+	fassignS(frd, tmp);
+}
+
+:fabs.h frd,frs1S is frd & frs1S & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct3=0x2 & funct7=0x12 & op1519=op2024
+{
+	local tmp:4 = abs(frs1S);
+	fassignS(frd, tmp);
+}
+
+:fmadd.h frd,frs1S,frs2S,frs3S,FRM is frs1S & frd & frs2S & FRM & frs3S & op0001=0x3 & op0204=0x0 & op0506=0x2 & op2526=0x2
+{
+	local tmp:4 = (frs1S f* frs2S) f+ frs3S;
+	fassignS(frd, tmp);
+}
+
+:fnmadd.h frd,frs1S,frs2S,frs3S,FRM is frs1S & frd & frs2S & FRM & frs3S & op0001=0x3 & op0204=0x3 & op0506=0x2 & op2526=0x2
+{
+	local tmp:4 = (f- (frs1S f* frs2S)) f- frs3S;
+	fassignS(frd, tmp);
+}
+
+:fmsub.h frd,frs1S,frs2S,frs3S,FRM is frs1S & frd & frs2S & FRM & frs3S & op0001=0x3 & op0204=0x1 & op0506=0x2 & op2526=0x2
+{
+	local tmp:4 = (frs1S f* frs2S) f- frs3S;
+	fassignS(frd, tmp);
+}
+
+:fnmsub.h frd,frs1S,frs2S,frs3S,FRM is frs1S & frd & frs2S & FRM & frs3S & op0001=0x3 & op0204=0x2 & op0506=0x2 & op2526=0x2
+{
+	local tmp:4 = (f- (frs1S f* frs2S)) f+ frs3S;
+	fassignS(frd, tmp);
+}
+
+:fcvt.w.h rdW,frs1S,FRM is frs1S & FRM & rdW & op0001=0x3 & op0204=0x4 & op0506=0x2 & funct7=0x62 & op2024=0x0
```

Comment:
https://github.com/NationalSecurityAgency/ghidra/pull/6492/files#diff-f74ddc93319f068d960c0557be5af9198befd17809b4144ba1f50653c199766eL44

Reply:
Does it make sense to not using rdW Here as well?

Reply:
Thanks - I'll take a look at this tomorrow.  I want to look at how qemu handles similar riscv instructions.

Have you any thoughts on how the semantics of *vector* floating point conversion instructions should be handled?  Element size is passed as often-changed SEL context variables, not in the instructions themselves.

Qemu source code spends about 24K lines of C to describe RISCV instruction semantics.  I'm not sure how much sleigh pcode that implies, or how to finish what we're starting.